tests/testthat/test-parsing-factors.R

```diff
@@ -17,8 +17,39 @@ test_that("warning if value not in levels", {
 })
 
 test_that("NAs silently passed along", {
-  x <- parse_factor(c("a", "b", "NA"), levels = c("a", "b"))
+  x <- parse_factor(c("a", "b", "NA"), levels = c("a", "b"), include_na = FALSE)
   expect_equal(n_problems(x), 0)
   expect_equal(x, factor(c("a", "b", NA)))
 })
 
+test_that("levels = NULL (497)", {
+  x <- parse_factor(c("a", "b", "c", "b"), levels = NULL)
+
+  expect_equal(n_problems(x), 0)
+  expect_equal(x, factor(c("a", "b", "c", "b")))
+})
+
+test_that("NAs included in levels if desired", {
+  x <- parse_factor(c("NA", "b", "a"), levels = c("a", "b", NA), include_na = TRUE)
+  expect_equal(x, factor(c("NA", "b", "a"), exclude = NULL))
+
+  expect_warning(x <- parse_factor(c("NA", "b", "a"), levels = c("a", "b"), include_na = TRUE))
```

Comment:
Ok thinking about this more I think you are right, `include_na = TRUE` now will include the NA level if there is an NA in the input and it is not in the levels.